.github/workflows/backend.yml

```diff
@@ -0,0 +1,445 @@
+            -v $PWD/config/symbolicator/:/etc/symbolicator \
+            --network host \
+            --name symbolicator \
+            us.gcr.io/sentryio/symbolicator:nightly \
+            run -c /etc/symbolicator/config.yml
+          docker ps -a
+
+      - name: Run test
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make test-symbolicator
+
+      - name: Handle artifacts
+        uses: ./.github/actions/artifacts
+
+  typing:
+    needs: files-changed
+    name: backend typing
+    runs-on: ubuntu-20.04
+    timeout-minutes: 12
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: ./.github/actions/setup-python
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # Since we don't call the setup-sentry action we need to install libxmlsec1-dev
+      - name: Setup backend typing
+        if: needs.changes.outputs.backend == 'true'
+        env:
+          SENTRY_LIGHT_BUILD: 1
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends libxmlsec1-dev
+          python setup.py install_egg_info
+          pip install wheel # GitHub Actions does not have `wheel` installed by default
+          pip install -U -e ".[dev]"
+
+      - name: Run backend typing (${{ steps.setup.outputs.matrix-instance-number }} of ${{ strategy.job-total }})
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make backend-typing
+
+  # This check runs once all dependant jobs have passed
+  # It symbolizes that all required Backend checks have succesfully passed
+  # This check is the only required Github check
+  backend-required-check:
+    needs:
+      [
+        api-docs,
+        backend,
+        cli,
+        lint,
+        migration,
+        plugins,
+        relay,
+        snuba,
+        symbolicator,
+        typing,
+      ]
+    if: ${{ needs.changes.outputs.backend == 'true' }}
+    name: Backend
+    runs-on: ubuntu-20.04
+    steps:
+      - run: 'echo "All required checks have passed and we meet the requirements." '
+
+  # If there are no backend checks, we still want the required check to be satisfied
+  # In order to do this, we need to pretend that the real check run but hijacking the name
+  # This is the recommended approach by Github:
+  # https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks
+  fake-backend-required-check:
+    needs: files-changed
+    if: ${{ needs.changes.outputs.backend_any_type != 'true' }}
+    name: Backend
+    runs-on: ubuntu-20.04
+    steps:
+      - run: 'echo "The changed files do not require running the checks in this workflow." '
```

Comment:
This is redundant because we now only have a single job (`backend-required-check`), this hack was only necessary when we had matrices and we had to simulate all of the matrix names.